--- a/src/build/MetallibStep.zig
+++ b/src/build/MetallibStep.zig
@@ -44,6 +44,9 @@
         b,
         b.fmt("metal {s}", .{opts.name}),
     );
+    if (b.graph.environ_map.get("GHOSTTY_METAL_DEVELOPER_DIR")) |developer_dir| {
+        run_ir.setEnvironmentVariable("DEVELOPER_DIR", developer_dir);
+    }
     run_ir.addArgs(&.{ "/usr/bin/xcrun", "-sdk", sdk, "metal", "-o" });
     const output_ir = run_ir.addOutputFileArg(b.fmt("{s}.ir", .{opts.name}));
     run_ir.addArgs(&.{"-c"});
@@ -59,6 +62,9 @@
         b,
         b.fmt("metallib {s}", .{opts.name}),
     );
+    if (b.graph.environ_map.get("GHOSTTY_METAL_DEVELOPER_DIR")) |developer_dir| {
+        run_lib.setEnvironmentVariable("DEVELOPER_DIR", developer_dir);
+    }
     run_lib.addArgs(&.{ "/usr/bin/xcrun", "-sdk", sdk, "metallib", "-o" });
     const output_lib = run_lib.addOutputFileArg(b.fmt("{s}.metallib", .{opts.name}));
     run_lib.addFileArg(output_ir);
